datumaro/plugins/icdar_format/converter.py

```diff
@@ -0,0 +1,211 @@
+            f.write('\n'.join(l.name
+                for l in categories[AnnotationType.label])
+            )
+
+    def save_annotations(self, item, categories):
+        annotation = ''
+        for ann in item.annotations:
+            if ann.type == AnnotationType.bbox:
+                annotation += '%s %s %s %s' % (ann.x, ann.y,
+                    ann.x + ann.w, ann.y + ann.h)
+                if ann.label is not None:
+                    annotation += ' %s' % \
+                        categories[AnnotationType.label][ann.label].name
+            elif ann.type == AnnotationType.polygon:
+                annotation += ','.join(str(p) for p in ann.points)
+                if ann.label is not None:
+                    annotation += ',\"%s\"' % \
+                        categories[AnnotationType.label][ann.label].name
+            annotation += '\n'
+        self.annotations[item.id] = annotation
+
+    def write(self, path):
+        os.makedirs(path, exist_ok=True)
+        for item in self.annotations:
+            file = osp.join(path, 'gt_' + item + '.txt')
+            with open(file, 'w') as f:
+                f.write(self.annotations[item])
+
+    def is_empty(self):
+        return len(self.annotations) == 0
+
+class _TextSegmentationConverter():
+    def __init__(self):
+        self.annotations = {}
+        self.masks = {}
+
+    def save_categories(self, save_dir, categories):
+        pass
+
+    def save_annotations(self, item, categories):
+        annotation = ''
+        colormap = [(255, 255, 255)]
+        group = 0
+        for ann in item.annotations:
+            if ann.type == AnnotationType.mask:
+                char = ''
+                if ann.group != group:
+                    annotation += '\n'
+                    group += 1
+                if ann.attributes:
+                    if 'char' in ann.attributes:
+                        char = ann.attributes['char']
+                    if char == ' ':
+                        annotation += '#'
+                    if 'color' in ann.attributes:
+                        colormap.append(ann.attributes['color'])
+                        annotation += ' '.join(str(p)
+                            for p in ann.attributes['color'])
+                    else:
+                        annotation += '- - -'
+                    if 'center' in ann.attributes:
+                        annotation += ' '
+                        annotation += ' '.join(str(p)
+                            for p in ann.attributes['center'])
+                    else:
+                        annotation += ' - -'
+                image = ann.image
+                bbox = find_mask_bbox(image)
+                annotation += ' %s %s %s %s' % (bbox[0], bbox[1],
+                    bbox[0] + bbox[2], bbox[1] + bbox[3])
+                annotation += ' \"%s\"' % char
+                annotation += '\n'
+
+        masks = [a for a in item.annotations
```

Comment:
Probably, we can move this higher, and replace the first loop iterable with this.